test/e2e/performanceprofile/functests/utils/nodes/nodes.go

```diff
@@ -113,39 +114,43 @@ func GetNonPerformancesWorkers(nodeSelectorLabels map[string]string) ([]corev1.N
 	return nonPerformanceWorkerNodes, err
 }
 
-// GetMachineConfigDaemonByNode returns the machine-config-daemon pod that runs on the specified node
-func GetMachineConfigDaemonByNode(node *corev1.Node) (*corev1.Pod, error) {
+// getTestingDaemonPodByNode returns the testing daemon pod that runs on the specified node
+func getTestingDaemonPodByNode(node *corev1.Node) (*corev1.Pod, error) {
 	listOptions := &client.ListOptions{
-		Namespace:     testutils.NamespaceMachineConfigOperator,
+		Namespace:     testutils.TestingDaemonsetNamespace,
 		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node.Name}),
-		LabelSelector: labels.SelectorFromSet(labels.Set{"k8s-app": "machine-config-daemon"}),
+		LabelSelector: labels.SelectorFromSet(labels.Set{"name": testutils.TestingDaemonsetName}),
 	}
 
-	mcds := &corev1.PodList{}
-	if err := testclient.Client.List(context.TODO(), mcds, listOptions); err != nil {
+	pods := &corev1.PodList{}
+	if err := testclient.DataPlaneClient.List(context.TODO(), pods, listOptions); err != nil {
 		return nil, err
 	}
-
-	if len(mcds.Items) < 1 {
-		return nil, fmt.Errorf("failed to get machine-config-daemon pod for the node %q", node.Name)
+	if len(pods.Items) < 1 {
+		return nil, fmt.Errorf("failed to get testing daemon pod for the node %q", node.Name)
 	}
-	return &mcds.Items[0], nil
+	return &pods.Items[0], nil
 }
 
-// ExecCommandOnMachineConfigDaemon returns the output of the command execution on the machine-config-daemon pod that runs on the specified node
-func ExecCommandOnMachineConfigDaemon(ctx context.Context, node *corev1.Node, command []string) ([]byte, error) {
```

Comment:
Both `ExecCommandOnTestingDaemonPod` and `ExecCommandOnNode` have the same purpose.
Unless there is a very good reason to have both functions, I would merge them into one named:
`ExecCommandOnNode` which declare explicitly where the command is about to be running.

Reply:
@Tal-or 
`ExecCommandOnTestingDaemonPod` is being used as an underlying function for `ExecCommandOnNode`.
The difference between the two is that `ExecCommandOnNode` returns a string ([after some formatting](https://github.com/openshift/cluster-node-tuning-operator/blob/master/test/e2e/performanceprofile/functests/utils/nodes/nodes.go#L174)) rather than a bytes slice (`ExecCommandOnTestingDaemonPod`).
As throughout the repo,  developers have chosen to use both of them, I think we should keep both of these functions and just rename them appropriately. (Another option is to unify them to one function that returns []bytes and change all calls to target this func, following a call if needed to a util function that will do the string format that is left). But again the latter option would require some work for refactoring. 
WDYT?

Reply:
@Tal-or  I'll need to rename `ExecCommandOnNode` to `ExecCommandOnNodeToString` and `ExecCommandOnTestingDaemonPod` which returns a []byte to `ExecCommandOnNode`.
This however, may cause a slight confusion for developers that got used that  `ExecCommandOnNode` returns a string. but I think this is not a big deal. 
Another option could be to change `ExecCommandOnTestingDaemonPod` to `ExecCommandOnNodeToByteSlice`. I personally think we should go with the first option, even if it might cause a bit of confusion.

Reply:
since the name of the package is `nodes` I would rename `ExecCommandOnTestingDaemonPod` to `ExecCommand` and call it a day.
From the caller side it looks like:
`nodes.ExecCommand()` The package name provides a clear indication about where this command is about to be running.

Reply:
But what will be the name of the second func?
Anyways I was thinking that we should move these functions to node inspector package, as they rely on it running.
So it's going to be more of `nodeInspector.ExecCommand...`

Reply:
The node inspector is an implementation detail. you can call `nodeInspector.ExecCommand` from `nodes.ExecCommand`